superdesk/resource.py

```diff
@@ -152,6 +152,15 @@ def __init__(self, endpoint_name, app, service, endpoint_schema=None):
         app.register_resource(self.endpoint_name, endpoint_schema)
         superdesk.resources[self.endpoint_name] = self
 
+        for attr in self.__dir__():
+            for request_method in ['GET', 'POST', 'PATCH', 'PUT', 'DELETE']:
```

Comment:
it would be probably enough to do:
```
 for method in ['GET', 'POST', 'PATCH', 'PUT', 'DELETE']:
    if hasattr(self, 'pre_request_%s' % method.lower()):
        ...
```